src/engine/guidance/post_processing.cpp

```diff
@@ -528,12 +528,22 @@ bool isStaggeredIntersection(const RouteStep &previous, const RouteStep &current
     // Base decision on distance since the zig-zag is a visual clue.
     const constexpr auto MAX_STAGGERED_DISTANCE = 3; // debatable, but keep short to be on safe side
 
-    using namespace util::guidance;
+    const auto angle = [](const RouteStep &step) {
+        const auto &intersection = step.intersections.front();
+        const auto entry_bearing = intersection.bearings[intersection.in];
+        const auto exit_bearing = intersection.bearings[intersection.out];
+        return turn_angle(entry_bearing, exit_bearing);
+    };
+
+    // Instead of using turn modifiers (e.g. as in isRightTurn) we want to be more strict here.
+    // We do not want to trigger e.g. on sharp uturn'ish turns or going straight "turns".
+    // Therefore we use the turn angle to derive 90 degree'ish right / left turns.
+    // This more closely resembles what we understand as Staggered Intersection.
+    const auto is_right = [](const double angle) { return angle > 45 && angle < 135; };
+    const auto is_left = [](const double angle) { return angle > 225 && angle < 315; };
 
-    const auto left_right = isLeftTurn(previous.maneuver.instruction) && //
-                            isRightTurn(current.maneuver.instruction);
-    const auto right_left = isRightTurn(previous.maneuver.instruction) && //
-                            isLeftTurn(current.maneuver.instruction);
+    const auto left_right = is_left(angle(previous)) && is_right(angle(current));
+    const auto right_left = is_right(angle(previous)) && is_left(angle(current));
```

Comment:
should we also check for similar deviations or are combinations of sharp/slight still valid?
Have you looked at the combinations of extremes to see if they follow your idea of a staggered intersection?